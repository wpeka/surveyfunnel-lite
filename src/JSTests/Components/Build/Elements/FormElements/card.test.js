import React, { useCallback, useRef, useState } from "react";
import { HTML5Backend } from 'react-dnd-html5-backend'
import { DndProvider } from 'react-dnd'
import {shallow,mount} from 'enzyme';
import { render, screen, fireEvent, waitFor } from "@testing-library/react";

import Card from '../../../../../Components/Build/Elements/FormElements/Card';
import renderer from 'react-test-renderer';
let test1="";
const ele={
    id:"simple",
    text:"simple",
    index:"dfd",
    item:"sdf",
    moveCard:"dfd",
    editList:(item)=>{test1=item},
    deleteFromList:(item)=>{test1=""},
}
test('Card test', async () => {

    const onchange = ()=>{
        return;
    }
    const useDrag = jest.fn();


    window.React = React
    const component = renderer.create( <DndProvider backend={HTML5Backend}> <Card id={ele.id} text={ele.text} index={ele.index} item={ele} moveCard={ele.moveCard} editList={ele.editList} deleteFromList={ele.deleteFromList} /> </DndProvider> );
    let tree = component.toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
});

test("editCard function should execute",async ()=>{
    const component=mount( <DndProvider backend={HTML5Backend}> <Card id={ele.id} text={ele.text} index={ele.index} item={ele} moveCard={ele.moveCard} editList={ele.editList} deleteFromList={ele.deleteFromList} /> </DndProvider> );
    const btns=component.find('button');
    expect(btns.length).toBe(2);
    btns.first().simulate('click');
    expect(test1).toEqual(ele);
    btns.at(1).simulate('click');
    expect(test1).toEqual("");
    const div=component.find('.cardBox');
    div.simulate('mouseEnter');
})