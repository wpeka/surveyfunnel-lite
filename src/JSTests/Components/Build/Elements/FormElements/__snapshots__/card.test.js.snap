// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test 1`] = `
Array [
  " ",
  <div
    className="cardBox"
    style={
      Object {
        "cursor": "move",
        "opacity": 1,
      }
    }
  >
    <div
      className="surveyfunnel-lite-cardbox-title"
    >
      <img
        src="test-file-stub"
      />
      <h3 />
    </div>
    <div
      className="card-flex"
    >
      <button
        className="surveyfunnel-lite-cardBox-btn"
        onClick={[Function]}
      >
        <img
          src="test-file-stub"
        />
      </button>
      <button
        className="surveyfunnel-lite-cardBox-btn"
        onClick={[Function]}
      >
        <img
          src="test-file-stub"
        />
      </button>
    </div>
  </div>,
  " ",
]
`;
